prysmaticlabs/prysm
# Check attestation has pre state before retrieving it

**What type of PR is this?**
> Bug fix

**What does this PR do? Why is it needed?**
To verify the signature of an incoming unaggregated attestation over the wire, the beacon node will retrieve the pre state of block voted by that attestation.  The current order of operation is the following:
```
1. Get attestation pre state
2. Check if pre state for block exists, if not, place the attestation in pending queue
3. Verify attestation signature using pre state
```

The order of operation was obvious wrong before. This PR updates to the following:
```
1. Check if pre state for block exists, if not, place the attestation in pending queue
2. Get attestation pre state
3. Verify attestation signature using pre state
```

**Which issues(s) does this PR fix?**
Fixes part of #6148 

**Other notes for review**
Tested run time

cc: @farazdagi
